runtime-infra/pn-ipc.yaml

```diff
@@ -98,6 +139,166 @@ Parameters:
     Type: String
   LogsExporterPolicyArn:
     Type: String
+  AlbSecurityGroup:
+    Type: String
+  CognitoUserPoolArn:
+    Type: String
+    Default: ''
+  CognitoClientId:
+    Type: String
+    Default: ''
+  ConfidentialInfoAccountId:
+    Default: ''
+    Type: String
+  OpenSearchArn:
+    Default: ''
+    Type: String
+  OpenSearchEndpoint:
+    Type: String
+    Default: ''
+  HelpdeskAccountId:
+    Default: ''
+    Type: String
+  CdcKinesisStreamName:
+    Type: String
+  CdcKinesisStreamArn:
+    Type: String
+  CdcKinesisStreamKeyArn:
+    Type: String
+  LogsKinesisStreamName:
+    Type: String
+  LogsKinesisStreamArn:
+    Type: String
+  LogsKinesisStreamKeyArn:
+    Type: String
+  SafeStorageAccountId:
+    Type: String
+  ApiKeyTablesEncryptKeyId:
+    Type: String
+
+  VpcEgressSubnetsIds:
+    Type: String
+
+  ElasticacheEndpoint:
+    Type: String
+    Default: ''
+  ElasticacheSecurityGroup:
+    Type: String
+    Default: ''
+
+  CdnZoneId:
+    Type: String
+    Default: '-'  
+  PortalePaCertificateArn:
+    Type: String
+    Default: '-'
+  PortalePfCertificateArn:
+    Type: String
+    Default: '-'    
+  PortalePfLoginCertificateArn:
+    Type: String
+    Default: '-'    
+  LandingCertificateArn:
+    Type: String
+    Default: '-' 
+  PortalePgCertificateArn:
+    Type: String
+    Default: '-' 
+  PortaleStatusCertificateArn:
+    Type: String
+    Default: '-' 
+  PortaleHelpdeskCertificateArn:
+    Type: String
+    Default: '-'
+  PortalePaDomain:
+    Type: String
+    Default: '-' 
+  PortalePfDomain:
+    Type: String
+    Default: '-' 
+  PortalePfLoginDomain:
+    Type: String
+    Default: '-' 
+  LandingDomain:
+    Type: String
+    Default: '-' 
+  PortalePgDomain:
+    Type: String
+    Default: '-' 
+  PortaleStatusDomain:
+    Type: String
+    Default: '-' 
+  PortaleHelpdeskDomain:
+    Type: String
+    Default: '-'
+  HubLoginDomain: 
+    Type: String
+    Default: '-'
+  ReactAppUrlApi:
+    Type: String
+    Default: '-' 
+  TokenExchangeLambdaEnvironmentIssuer:
+    Type: String
+    Description: Session token issuer
+  TokenExchangeLambdaEnvironmentAllowedIssuer:
+    Type: String
+    Description: 'Comma separated list of external authentication token issuers 
+                  recognized by Piattaforma Notifiche (SelfCare and SpidHub)'
+  TokenExchangeLambdaEnvironmentAllowedOrigin:
+    Type: String
+    Description: 'Comma separated list of websites allowed to use "Piattaforma 
+                 Notifiche" back-ends API'
+  TokenExchangeLambdaEnvironmentJwksMapping:
+    Type: String
+    Description: 'Mapping tra issuer e jwks url non standard in formato json'
+  TokenExchangeLambdaEnvironmentAcceptedAudience:
+    Type: String
+    Description: Comma separated list of token audience accepted
+  TokenExchangeLambdaEnvironmentAudience:
+    Type: String
+    Description: Audience value for token generation
+  ApikeyAuthorizerV2PDNDAudience:
+    Type: String
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded API Gateway API Key. Hardcoded secrets can allow attackers to bypass authentication methods and perform malicious actions. Possible remediation approaches are:

	 1. Manage secret data using environment variables:
 		 a. Identify the secret data and store that data in environment variables on your local and production server. The setting process may vary depending on your OS and environment.
		 b. Replace hard-coded secrets with references to environment variables, e.g., `password = os.environ.get(&#39;PASSWORD&#39;)`.
	 2. Use AWS Secrets Manager to store, rotate, monitor, and control access to secrets. To retrieve secrets, you can replace hardcoded secrets in applications with a call to Secrets Manager APIs. To use Secrets Manager:
		 a. Visit [Secrets Manager](https://aws.amazon.com/secrets-manager/) on the AWS Management Console.
		 b. Choose the secret type on the console page and follow the instructions.
		 c. Use the code samples suggested in the console to retrieve the secret in your application.